src/main/java/org/opensearch/flowframework/FlowFrameworkPlugin.java

```diff
@@ -93,10 +96,19 @@ public Collection<Object> createComponents(
         Settings settings = environment.settings();
         this.clusterService = clusterService;
         flowFrameworkFeatureEnabledSetting = new FlowFrameworkFeatureEnabledSetting(clusterService, settings);
-
+        FlowFrameworkMaxRequestRetrySetting flowFrameworkMaxRequestRetrySetting = new FlowFrameworkMaxRequestRetrySetting(
+            clusterService,
+            settings
+        );
         MachineLearningNodeClient mlClient = new MachineLearningNodeClient(client);
         FlowFrameworkIndicesHandler flowFrameworkIndicesHandler = new FlowFrameworkIndicesHandler(client, clusterService);
-        WorkflowStepFactory workflowStepFactory = new WorkflowStepFactory(clusterService, client, mlClient, flowFrameworkIndicesHandler);
+        WorkflowStepFactory workflowStepFactory = new WorkflowStepFactory(
+            clusterService,
+            client,
+            mlClient,
+            flowFrameworkIndicesHandler,
+            flowFrameworkMaxRequestRetrySetting
```

Comment:
Instead of passing `flowFrameworkMaxRequestRetrySetting` here, we should pass `settings` because it might be needed later for other WorkflowSteps and make it generic.

Reply:
thats a good point, I'll change this so that the settings object itself is passed to the workflow step factory